File: frontends/Python/Mesh/writemesh.py
from numpy import *

def writemesh(p, t, filename, mode):

    nd,np = p.shape;
    nve,ne = t.shape;
    tmp = int64(array([nd, np, nve, ne]));

    if mode==0: # binary
        fileID = open(filename, "wb");
        tmp.astype('float64').tofile(fileID);
        p = p.flatten(order = 'F');
        p.astype('float64').tofile(fileID);
        t = t.flatten(order = 'F');
        t.astype('float64').tofile(fileID);
        fileID.close();
    else: # text
        fileID = open(filename, "w");
        savetxt(fileID, tmp.reshape((1,4)), fmt='%d');
        savetxt(fileID, p.T, fmt='%.16f');
        savetxt(fileID, t.T, fmt='%d');
        fileID.close();
    return 0;

File: frontends/Julia/Mesh/writemesh.jl
using DelimitedFiles

function writemesh(p, t, filename, mode)

nd,np = size(p);
nve,ne = size(t);
tmp = [nd, np, nve, ne];

fileID = open(filename, "w");
if mode==0 # binary
    write(fileID,Float64.(tmp));
    write(fileID,Float64.(p));
    write(fileID,Float64.(t));
else # text
    writedlm(fileID, tmp');
    writedlm(fileID, p');
    writedlm(fileID, t');
end
close(fileID);

end
